### PR TITLE
fix(gatsby-plugin-image): Fix handling of sizes prop in SSR

### PR DESCRIPTION
**Pull request by ascorbic**
The sizes prop was being incorrectly handled in SSR. It should be using the property of the fallback image, not a top-level sizes prop.

### CODE REVIEW

File: packages/gatsby-plugin-image/src/components/gatsby-image.server.tsx [hunk 1]
```diff
@@ -49,7 +49,6 @@ export const GatsbyImage: FunctionComponent<GatsbyImageProps> = function GatsbyI
     layout,
```
Comment (pvdz) [inline packages/gatsby-plugin-image/src/components/gatsby-image.server.tsx:49]:
```suggestion
    layout,
    fallback,
```

File: packages/gatsby-plugin-image/src/components/gatsby-image.server.tsx [hunk 2]
```diff
@@ -65,7 +64,7 @@ export const GatsbyImage: FunctionComponent<GatsbyImageProps> = function GatsbyI
   }
   if (images.fallback) {
     cleanedImages.fallback = {
-      src: images.fallback.src,
+      ...images.fallback,
```
Comment (pvdz) [inline packages/gatsby-plugin-image/src/components/gatsby-image.server.tsx:67]:
```suggestion
  if (fallback) {
    cleanedImages.fallback = {
      ...fallback,
```

**Review by pvdz — DISMISSED** [review]
Deja-vu ;)

The spread will pass on an unknown set of properties. I'd prefer to see them set explicitly. But not blocking you on this. Feel free to ignore the suggestion.

**Review by laurieontech — COMMENTED** [review]
In general this looks good. I'd like to include an e2e test with sizes as a prop since that case isn't covered and this feels like a good opportunity.

File: e2e-tests/gatsby-static-image/src/pages/constrained.js [hunk 3]
```diff
@@ -5,17 +5,15 @@ import Layout from "../components/layout"
 
 const FluidPage = () => (
```
Comment (laurieontech) [inline e2e-tests/gatsby-static-image/src/pages/constrained.js:6]:
```suggestion
const ConstrainedPage = () => (
```

**Review by laurieontech — COMMENTED** [review]
Thanks for the test additions. Is it still the case that <GatsbyImage sizes={}> with some sizes data will override the automatic sizes? That's the main use case I'm worried we waren't testing.

**Comment by ascorbic** [comment]
@laurieontech Ah, gotcha. Added a test for that

**Review by laurieontech — APPROVED** [review]
Looks great!

**Pull request merged**